stdlib/public/core/Sequence.swift

```diff
@@ -1126,6 +1147,31 @@ extension Sequence where
     }
     return AnySequence(result)
   }
+  
+  /// Returns a subsequence by skipping elements while `predicate` returns
+  /// `true` and returning the remaining elements.
+  ///
+  /// - Parameter while: A closure that takes an element of the
+  ///   sequence as its argument and returns `true` if the element should
+  ///		be skipped or `false` if it should be included. Once the predicate
+  ///		returns `false` it will not be called again.
+  public func drop(
+    while predicate: @noescape (Iterator.Element) throws -> Bool
+  ) rethrows -> AnySequence<Iterator.Element> {
+    var iterator = makeIterator()
+    var nextElement: Iterator.Element? = iterator.next()
+    repeat {
+      nextElement = iterator.next()
+    } while try nextElement.flatMap(predicate) == true
+
+    return AnySequence {
+      AnyIterator {
```

Comment:
I was going to ask why the closure is more expensive than a class but your point about multiple levels of wrappers makes sense; I'll work on this.